### ISSUE TITLE
Option to not force fullscreen to be topmost (alt+tab issue)

### ISSUE THREAD

**Issue by rebane2001** [pyglet/pyglet#566]
Right now, if you create a fullscreen window in pyglet, it sets the HWND_TOPMOST flag for it, which means it is always on top of everything else. This means that you cannot use alt+tab to switch between the pyglet window and other programs, which is bad if you want to do *anything* else (chatting, looking up things on the web etc) while a pyglet game is running.

My proposed solution would be to add an argument to enable fullscreen without setting the HWND_TOPMOST flag, maybe something like window(topmost=...), where the value can be set to True, False, or auto (current behaviour).

As a temporary workaround, monkey patching can be used:
```py
import pyglet.libs.win32.constants
pyglet.libs.win32.constants.HWND_TOPMOST = pyglet.libs.win32.constants.HWND_NOTOPMOST

# import only after patching
import pyglet
```

Note that this post applies to Windows ([pyglet.window.win32](https://github.com/pyglet/pyglet/blob/pyglet-1.5-maintenance/pyglet/window/win32/__init__.py)), I have not tested this behaviour on other platforms.

**Comment by rebane2001** [comment]
I tried fullscreen on a Mac and it seems to also have the same behaviour.

**Comment by benmoran56** [comment]
Thanks for opening the ticket.
If you set `window.set_exclusive_keyboard(False)`, does it still prevent alt+tab?

**Comment by rebane2001** [comment]
I'm not at a computer at the moment, but iirc that was the case. I could actually alt+tab into other programs, but they wouldn't be visible since the pyglet window forced itself to be the topmost.

**Comment by LennyPhoenix** [comment]
I'd be willing to try and implement this if I get the time this week, would a kwarg at `set_fullscreen` or `__init__` be better?

**Comment by benmoran56** [comment]
@LennyPhoenix 

In addition to fullscreen, we're also setting HWND_TOPMOST when the Window style is set to "overlay". 
That part should remain as it is,  but it does seem like HWND_NOTOPMOST should be the default otherwise. 

I was thinking about maybe adding a class variable to allow the user to specifically set it, but maybe that's not necessary.

**Comment by LennyPhoenix** [comment]
It would be nice to be able to control the flags individually, but that's probably for another PR. I'll start working on this tomorrow once I have access to a Windows machine :+1:

**Comment by benmoran56** [comment]
Closing this, as this was already addressed by Lenny.
